exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/ResultRow.kt

```diff
@@ -165,16 +165,26 @@ class ResultRow(
         fun createAndFillDefaults(columns: List<Column<*>>): ResultRow =
             ResultRow(columns.withIndex().associate { it.value to it.index }).apply {
                 columns.forEach {
-                    val value = when {
-                        it.defaultValueFun != null -> it.defaultValueFun!!()
-                        it.columnType.nullable -> null
-                        else -> NotInitializedValue
-                    }
-                    setInternal(it, value)
+                    setInternal(it, it.defaultValueOrNotInitialized())
                 }
             }
     }
 
+    private fun <T> Column<T>.defaultValueOrNotInitialized(): Any? {
+        return when {
+            defaultValueFun != null -> when {
+                columnType is ColumnWithTransform<*, *> -> {
+                    (columnType as ColumnWithTransform<Any, Any>).unwrapRecursive(defaultValueFun!!())
+                }
+
+                else -> defaultValueFun!!()
+            }
+
+            columnType.nullable -> null
+            else -> NotInitializedValue
+        }
```

Comment:
It's minor and just my opinion, but if you don't anticipate the second `when` block having more branches, it looks cleaner and logic is easier to read with single block:
```kt
return when {
    defaultValueFun != null && columnType is ColumnWithTransform<*, *> -> {
        (columnType as ColumnWithTransform<Any, Any>).unwrapRecursive(defaultValueFun!!())
    }
    defaultValueFun != null -> defaultValueFun!!()
    columnType.nullable -> null
    else -> NotInitializedValue
}
```

Reply:
It is Idea's default auto correction, and we have no explicit value in `.editorconfig` for it, so I'd add it there also.
I will add a line `ij_kotlin_line_break_after_multiline_when_entry = false` (according to the comment [KTIJ-21944](https://youtrack.jetbrains.com/issue/KTIJ-21944/Code-style-does-not-honour-setting-for-Minimum-blank-lines-around-when-branches-with#focus=Comments-27-6450223.0-0)) to the `.editorconfig` with this PR.